src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,812 @@
+
+                Handle_Behavior_Z(obj, physics, goal_pos);
+            } else {
+                if (m_template->m_appearance == LOCO_WINGS) {
+                    Set_Flag(IS_BRAKING, false);
+                }
+
+                bool breaking = obj->Get_Status_Bits().Test(OBJECT_STATUS_IS_BRAKING);
+                physics->Set_Turning(TURN_NONE);
+
+                if (Get_Allow_Motive_Force_While_Airborne() || !b) {
+                    switch (m_template->m_appearance) {
+                        case LOCO_LEGS_TWO:
+                            Move_Towards_Position_Legs(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        case LOCO_WHEELS_FOUR:
+                        case LOCO_MOTORCYCLE:
+                            Move_Towards_Position_Wheels(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        case LOCO_TREADS:
+                            Move_Towards_Position_Treads(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        case LOCO_HOVER:
+                            Move_Towards_Position_Hover(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        case LOCO_THRUST:
+                            Move_Towards_Position_Thrust(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        case LOCO_WINGS:
+                            Move_Towards_Position_Wings(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        case LOCO_CLIMBER:
+                            Move_Towards_Position_Climb(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                        default:
+                            Move_Towards_Position_Other(obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
+                            break;
+                    }
+                }
+
+                Handle_Behavior_Z(obj, physics, goal_pos);
+                obj->Set_Status(
+                    BitFlags<OBJECT_STATUS_COUNT>(BitFlags<OBJECT_STATUS_COUNT>::kInit, OBJECT_STATUS_IS_BRAKING),
+                    Get_Flag(IS_BRAKING));
+
+                if (breaking) {
+                    Coord3D newpos = *obj->Get_Position();
+
+                    if (obj->Is_KindOf(KINDOF_PROJECTILE)) {
+                        obj->Set_Status(
+                            BitFlags<OBJECT_STATUS_COUNT>(BitFlags<OBJECT_STATUS_COUNT>::kInit, OBJECT_STATUS_IS_BRAKING),
+                            true);
+                        path_dist = GameMath::Sqrt(x * x + y * y + z * z);
+                        float magnitude = physics->Get_Velocity_Magnitude();
+
+                        if (magnitude < 0.33333334f) {
```

Comment:
0.33333334 is 100/30, there should be some define or const somewhere for this
30 corresponds to the frame rate so we should fix these, so we know where to find frame rate specific code

Reply:
fixed.